sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/ExternalPythonTransform.java

```diff
@@ -33,25 +40,169 @@
 import org.apache.beam.sdk.values.POutput;
 import org.apache.beam.sdk.values.Row;
 import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.vendor.bytebuddy.v1_11_0.net.bytebuddy.ByteBuddy;
```

Comment:
Using ByteBuddy to create a class dynamically from which we can infer the schema seems overkill if we have a set of (type, name)s. Could we use StaticSchemaInference.fieldFromType directly instead?

Reply:
Do you see a downside for this approach ? I would like to use an existing proven utility here (POJO -> schema) instead trying to recursively and manually generating the schema here. We could end up doing an implementation similar to what's already done by POJO -> schema conversion and might be hard to handle all corner cases that are already handled by existing conversion.

If you don't see a downside can we try to get the API with this implementation in and look for an alternate implementation later ? (for example, if the second implementation is more efficient somehow)

Reply:
The downside is that it's way more complicated than need be, and likely more brittle too (using bytebuddy, invoking class loaders, etc.) It also accumulates garbage in global state (e.g. the classloader, schema registry). The POJO -> schema conversions are complicated because they handle all kinds of variants of how fields are named, annotations, private members vs autovalues vs getter/setters, etc. which are not relevant here. 

What about

```
  Schema schemaFromMap(Map<String, Object> fields) {
    Schema.Builder builder = Schema.builder();
    for (Map.Entry<String, Object> entry : fields.entrySet()) {
      Schema.builder().addField(
          entry.getKey(),
          StaticSchemaInference.fieldFromType(
              TypeDescriptor.of(entry.getValue().getClass()),
              JavaFieldSchema.JavaFieldTypeSupplier.INSTANCE));
    }
    return builder.build();
  }
```

Reply:
Updated to use provided field types directly instead of using ByteBuddy.